cmd/docker/docker.go

```diff
@@ -308,9 +308,8 @@ func runDocker(ctx context.Context, dockerCli *command.DockerCli) error {
 		return err
 	}
 
-	mp := dockerCli.MeterProvider(ctx)
-	defer mp.Shutdown(ctx)
-	otel.SetMeterProvider(mp)
+	mp := dockerCli.MeterProvider()
+	defer mp.(command.MeterProvider).Shutdown(ctx)
```

Comment:
Is there a way we can move this shutdown into the general workflow of the plugins to happen automatically? I'm a bit uncomfortable with this typecast here. It seems like we either need to add a `Shutdown()` method to the `dockerCli` struct itself or find some way to have it happen automatically. We can register the shutdown methods with the `dockerCli` struct when we initialize the tracer and meter provider.

If we add a `Shutdown` method though, we need to modify every plugin to use it. It would be nice if we could wire this into cobra so it happens automatically when the CLI is exiting.

Reply:
This particular code that you're commenting on isn't relevant for the plugin workflow, this is part of the main CLI binary execution.

Reply:
Also, would a safe cast (with an error/debug log if we failed to cast) look better to you here? It doesn't worry me since this is only inside the CLI codebase and no external import would change that `mp` before we cast, but I wouldn't mind a safer approach I guess.

Reply:
+1 on a safer cast (can still print a big red error though).

If in some case it turns out that the cast fails, I'd much rather just miss an OTEL flush than get a panic before the hook even gets executed.